src/app/course-page/services/show-params.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { ItemCourseService } from './item-course.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ShowParamsService {
```

Comment:
You don't need this class, please change fetchItems to be able to accept startIndex and count parameters.

Reply:
Removed this service